imageprocessor/views.py

```diff
@@ -20,15 +25,29 @@ def tagged_pictures(request):
 
 def classify(request):
 	context = {}
-	form = ImageForm(request.POST or None)
+	form = ImageForm(request.POST or None, request.FILES or None)
 	if request.method == 'POST':
 		try:
-			image_file = request.FILES.get('file')
+			image_file = request.FILES['file']
 			image = Image.open(image_file)
 			context['tags'] = detect(image)
+			new_image = form.save()
+			new_image.save()
+			context['new_image'] = new_image
 			return render(request, 'output.html', context)
 		except OSError as err:
 			context['form'] = form
 			return render(request, 'input.html', context)
 	context['form'] = form
 	return render(request, 'input.html', context)
+
+def register(request):
```

Comment:
Can you add tests for this?

Reply:
imageprocessor/tests.py line 32 tests this function.